test/integration/integration_test.go

```diff
@@ -169,10 +170,10 @@ func FeatureContext(s *godog.Suite) {
 		getRoutingUrlAndVerifyHTTPResponse)
 
 	// steps for verifying config file content
-	s.Step(`JSON config file "([^"]*)" (contains|does not contain) key "(.*)" with value "(.*)"$`,
-		configContains)
-	s.Step(`JSON config file "([^"]*)" (contains|does not contain) key "(.*)"(.*)$`,
-		configContains)
+	s.Step(`^JSON config file "([^"]*)" (contains|does not contain) key "(.*)" with value "(.*)"$`,
+		jsonConfigContains)
+	s.Step(`^JSON config file "([^"]*)" (has|does not have) key "(.*)" with value assigned$`,
```

Comment:
> has key or or does not have key... the value would have to be irrelevant here...
even "" is a value...

whether an empty value matters would be implementation and Viper dependent. So there might be and probably is a difference here. However, in this context I think we are talking about whether a given key exists or not (assuming that you mange keys properly via config set|unset. So +1 for removing "with value assigned".